examples/lp/test_lp.cu

```diff
@@ -0,0 +1,131 @@
+    * @param  parameters Command line parameters
+    * @param  v,s,val    Place holders for type deduction
+    * \return cudaError_t error message(s), if any
+    */
+   template <typename VertexT,  // Use int as the vertex identifier
+             typename SizeT,    // Use int as the graph size type
+             typename ValueT>   // Use int as the value type
+   cudaError_t
+   operator()(util::Parameters &parameters, VertexT v, SizeT s, ValueT val) {
+     typedef typename app::TestGraph<VertexT, SizeT, ValueT,
+                                     graph::HAS_CSR | graph::HAS_CSC>
+         GraphT;
+     typedef typename GraphT::VertexT LabelT;
+     typedef typename GraphT::CsrT CsrT;
+ 
+     cudaError_t retval = cudaSuccess;
+     util::CpuTimer cpu_timer;
+     GraphT graph;  // graph we process on
+ 
+     cpu_timer.Start();
+     GUARD_CU(graphio::LoadGraph(parameters, graph));
+     cpu_timer.Stop();
+     parameters.Set("load-time", cpu_timer.ElapsedMillis());
+ 
+     GUARD_CU(app::Set_Srcs(parameters, graph));
+     LabelT **ref_labels = NULL;
+     int num_srcs = 0;
+     bool quick = parameters.Get<bool>("quick");
+     // compute reference CPU BFS solution for source-distance
+     if (!quick) {
+       bool quiet = parameters.Get<bool>("quiet");
+       std::string validation = parameters.Get<std::string>("validation");
+       util::PrintMsg("Computing reference value ...", !quiet);
+       std::vector<VertexT> srcs = parameters.Get<std::vector<VertexT> >("srcs");
+       num_srcs = srcs.size();
+       SizeT nodes = graph.nodes;
+       ref_labels = new LabelT *[num_srcs];
+       for (int i = 0; i < num_srcs; i++) {
+         ref_labels[i] = (LabelT *)malloc(sizeof(LabelT) * nodes);
+         auto src = srcs[i];
+         VertexT num_iters = 0;
+         util::PrintMsg("__________________________", !quiet);
+         float elapsed = app::lp::CPU_Reference(graph, src, quiet, false,
+                                                 ref_labels[i], (VertexT*)NULL, num_iters);
+         util::PrintMsg("--------------------------\nRun " + std::to_string(i) +
+                            " elapsed: " + std::to_string(elapsed) +
+                            " ms, src = " + std::to_string(src) +
+                            ", #iterations = " + std::to_string(num_iters),
+                        !quiet);
+       }
+     }
+ 
+     std::vector<std::string> switches{
+         "mark-pred", "advance-mode", 
+         "idempotence"};
+     GUARD_CU(app::Switch_Parameters(
+         parameters, graph, switches,
+         [ref_labels](util::Parameters &parameters, GraphT &graph) {
+           return app::lp::RunTests(parameters, graph, ref_labels);
+         }));
+ 
+     if (!quick) {
+       for (int i = 0; i < num_srcs; i++) {
+         free(ref_labels[i]);
+         ref_labels[i] = NULL;
+       }
+       delete[] ref_labels;
+       ref_labels = NULL;
+     }
+     return retval;
+   }
+ };
+ 
```

Comment:
This will be the main of your example program. This could be a place to add additional README-type information about your work, how to properly use it, certain assumptions you made, or things like that.